src/client/datascience/data-viewing/dataViewerFactory.ts

```diff
@@ -46,4 +60,29 @@ export class DataViewerFactory implements IDataViewerFactory, IAsyncDisposable {
         }
         return result;
     }
+
+    private updateOpenDataViewers(viewer: IDataViewer) {
+        this.knownViewers.delete(viewer);
+    }
+
+    private async updateViewStateContext() {
+        // A data viewer's view state has changed. Look through our known viewers to see if any are active
+        let hasActiveViewer = false;
+        this.knownViewers.forEach((viewer) => {
+            if (viewer.active) {
+                hasActiveViewer = true;
+            }
+        });
+        await this.viewContext.set(hasActiveViewer);
+    }
+
+    private refreshDataViewer() {
+        // Find the data viewer which is currently active
+        for (const viewer of this.knownViewers) {
+            if (viewer.active) {
```

Comment:
Not sure viewer.active is all that accurate unfortunately. I've had problems with interactive window not turning this to false when being hidden.

Reply:
FWIW I implemented `viewer.active` as a getter whose value is `webPanel.isActive()`, and that in turn is directly equal to the vscode webview panel's `active` property, so this is slightly different from what the interactive window has. I implemented it this way because I noticed that the viewstate in the webPanelHost was wrong. Even after being disposed the viewstate.active property was still `true` because `onDidChangeViewState` doesn't fire when vscode webview panels are disposed. It's possible that is same reason that the interactive window's `active` property is wrong.